common/lib/xmodule/xmodule/x_module.py

```diff
@@ -770,9 +770,14 @@ def public_view(self, _context):
         )
 
         if self.display_name:
-            display_text = DEFAULT_PUBLIC_VIEW_MESSAGE %self.display_name
+            display_text = _(
+                "{display_name} is only accessible to enrolled learners."
+                " Sign in or register, and enroll in this course to view it"
+                ).format(
+                    display_name=self.display_name
+                )
         else:
-            display_text = DEFAULT_PUBLIC_VIEW_MESSAGE %"This content"
+            display_text = DEFAULT_PUBLIC_VIEW_MESSAGE
 
```

Comment:
Thanks. DEFAULT_PUBLIC_VIEW_MESSAGE will also need to be in a `_()` so that it gets translated. Also second sentence is missing a full-stop at the end. In case you haven't seen it, [i18n docs](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/internationalization/i18n.html) should cover all the details.

Reply:
Thanks @symbolist . I've corrected it now.